src/utils/CredentialUtils.ts

```diff
@@ -0,0 +1,43 @@
+import { ICredential } from '@sphereon/ssi-types'
+import { ICredentialStatus } from '@sphereon/ssi-types/src/types/vc'
+
+import { CredentialStatusEnum, ICredentialSummary } from '../@types'
+
+import DateUtils from './DateUtils'
+
+export const getCredentialStatus = (credential: ICredential | ICredentialSummary) : CredentialStatusEnum => {
+    return (
+      isRevoked(credential?.credentialStatus) ||
+      isExpired(credential?.expirationDate) ||
+      CredentialStatusEnum.VALID
+    )
+  }
+
+export const isRevoked = (credentialStatus: CredentialStatusEnum | ICredentialStatus | undefined) : CredentialStatusEnum | undefined => {
+    if (credentialStatus === CredentialStatusEnum.REVOKED) {
+      return CredentialStatusEnum.REVOKED
+    }
+    // FIXME if credentialStatus is of type ICredentialStatus then it may have following structure. For this we may need to make a network call to confirm the status.
+    // This should be done in a separate ticket.
+    // {
+    //  id: 'https://revocation-sphereon.sphereon.io/services/credentials/wallet-dev#2022021400',
+    //  type: 'RevocationList2022021401Status',
+    //  revocationListIndex: '2022021402',
+    //  revocationListCredential: 'https://revocation-sphereon.sphereon.io/services/credentials/wallet-dev#2022021400',
+    // }
+  }
+
+export const isExpired = (expirationDate: string | number | undefined) : CredentialStatusEnum | undefined => {
```

Comment:
i would expect this method to return if a credential is expired (boolean) and then use that value to return a certain status. Then we can reuse this function a lot more.

Same for the isRevoked